message.go

```diff
@@ -1,49 +1,43 @@
 package proton
 
 import (
+	"bytes"
 	"context"
 	"fmt"
 	"runtime"
 	"strconv"
 
-	"github.com/bradenaw/juniper/iterator"
 	"github.com/bradenaw/juniper/parallel"
-	"github.com/bradenaw/juniper/stream"
 	"github.com/bradenaw/juniper/xslices"
 	"github.com/go-resty/resty/v2"
 )
 
 const maxMessageIDs = 1000
 
-func (c *Client) GetFullMessage(ctx context.Context, messageID string) (FullMessage, error) {
+func (c *Client) GetFullMessage(ctx context.Context, messageID string, scheduler Scheduler, storageProvider AttachmentAllocator) (FullMessage, error) {
 	message, err := c.GetMessage(ctx, messageID)
 	if err != nil {
 		return FullMessage{}, err
 	}
 
-	attData, err := c.attPool().ProcessAll(ctx, xslices.Map(message.Attachments, func(att Attachment) string {
+	attDataBuffers, err := scheduler.Schedule(ctx, xslices.Map(message.Attachments, func(att Attachment) string {
 		return att.ID
-	}))
+	}), storageProvider, func(s string, buffer *bytes.Buffer) error {
+		return c.GetAttachmentInto(ctx, s, buffer)
+	})
 	if err != nil {
 		return FullMessage{}, err
 	}
 
 	return FullMessage{
-		Message: message,
-		AttData: attData,
+		Message:        message,
+		AttDataBuffers: attDataBuffers,
```

Comment:
Do you really need both the underlying buffers and the buffers' bytes? Isn't that potentially a mem duplication?

Reply:
There is no memory duplication here, but this is no longer required at the moment. Removed.